### DISCUSSION TITLE
CFWheels Help

### DISCUSSION THREAD

**Discussion by cfcoder** [wheels-dev/wheels#1379] [General]
As a relatively new CFWheels framework user, I'm finding the docs to be rather deficient in explaining some aspects of using Wheels. I post questions on this discussion group and occasionally get responses. Thank you to those who have responded.  But my list of questions keeps growing and the few of them I have posted in this group have gone unanswered.  I have a project to complete and waiting days to weeks for answers is not helpful.

Is there another better discussion group somewhere that has users with CFWheels experience where I can get more rapid responses?
It appears this discussion group is monitored occasionally by maybe 1-3 people who perhaps manage the site, with few others visiting and contributing.

I'm frustrated and wondering if CFWheels is the right framework for a new project, as it appears few new users are using it.

**Comment by dbelanger** [comment]
You're in the right spot for CFWheels help however ours is a small Open Source community and we can't always answer as soon as you'd like us too.  This is the way of pretty much all smaller Open Source communities.
What I love about Open Source docs is that anyone can contribute to them.  If you've learned something from one of our members that you feel should be in the docs, please contribute that information to the docs as you'll be helping out the next person who takes up the framework.
As for people using the framework, you are again correct in that we have a smaller user base than ColdBox (the leading CF framework with a much larger community) however I think you'll agree that CFWheels has a far easier learning curve.

**Reply by cfcoder** [reply]
At this point, I have questions, not answers. I will keep your request in mind when I think I know what should be added to the docs. Best I could offer for now is posts that request explanations for this and that.  For example, I find no mention of CF queries of queries support.  Is it natively supported or is that left for custom queries to be built into my models? There are other questions, this was just an example.

**Reply by chapmandu** [reply]
Hi Bruce, if you wanted to arrange a Zoom call that suits both our timezones we may be able to screen share and work through some of your questions.

**Reply by cfcoder** [reply]
@chapmandu Wow, that's very nice of you to offer!  What TZ are you in? I'm in the US PST. Time-wise, I'd be pretty flexible between my 8 am to 7pm PST.  I'd put together a list of questions to go over so that I don't waste your time. Just let me know what works for you.

**Comment by adamcameron** [comment]
> I'm frustrated and wondering if CFWheels is the right framework for a new project, as it appears few new users are using it.

TBH, I would not even start a new project with CFWheels even if I was a seasoned CFWheels dev. I would *definitely* not invest my time learning CFWheels from scratch for a new project in 2024.

CFWheels is mostly in maintenance mode. I'll get pushback on that comment from certain quarters, but the reality is only few ppl are doing a small amount of maintenance on it still (I am impressed by their... dedication), but I suspect this is largely because of their own existing time/effort investment in the platform, and wanting it to keep ticking over for their existing work. This is not the kind of effort that really spells "future" for the platform.

You are not in the situation of already being familiar with CFWheels, and I don't think it's knowledge that is really that useful for you to start learning now. I think you'd be far better off switching to ColdBox. ColdBox (and indeed the whole *Box ecosystem) has commercial backing by Ortus, and active *paid* developers working on their products. It also has continual revenue from support and other third-party commercial partners. It has a large, active and very helpful community. You will not be waiting long for your questions to be answered.

It is also very easy to get up and running on it, out of the gate.

Note: I do not have a dog in either fight here; I use CFWheels, and I do not use ColdBox. But I use other *Box products - TestBox, LogBox - and the support I get is very expedient and attentive.

This is also not a commentary on the technical merits of either platform. It is merely an observation about where I think one's effort would be better spent in 2024, starting a new project with a new framework, and thinking ahead to the future.

Good luck.

**Reply by cfcoder** [reply]
Thanks Adam. I inherited this project with a task to expand it. So actually I misspoke in saying it's a new project. The expansion of it is new. :) CFWheels is new for me for about two or three months now.  The existing portion of the project I found to not closely follow the guidelines of CFW.  So I've been trying to refactor it to more closely follow best practices as best I can determine from the brief (in places) documentation.  Given the timeline I'm under, I don't think switching to Coldbox is an option. I have worked with CB in the past (about 10 yrs ago) and yes, it's learning curve is steep but probably could have picked it up again with about equivalent effort as learning CFW from new. Oh well... thanks for your suggestion.

**Comment by bpamiri** [comment]
Bruce, I'm sorry you are feeling frustrated. One of my goals since I re-joined the project was to make CFWheels more accessible to new users. I fear I have failed at that. CFWheels has been around for a very long time and a tremendous amount of work has been done on the framework over the last 20 years by a handful of dedicated volunteers. I think the best way to embrace CFWheels is to embrace the whole open-source nature of it. 

Let me explain what I mean by that. As @dbelanger explained above, the nature of open source is that if you find something that needs changing, you are empowered to make that change for those coming up behind you. But in the real world, if you are finding a particular concept or page of the docs deficient, you may not have the knowledge, experience, or confidence to supply the change it needs. But trust your gut, if you are having difficulty with a topic, chances are others may find it difficult as well. So, what I humbly suggest is that you open an issue to have that topic or documentation page revised. This way we don't lose track of the need to improve the documentation on a topic.

Ultimately you have to make a decision that is right for you and your project, but I hope that CFWheels can be a part of that journey.

**Reply by cfcoder** [reply]
Thanks, Peter.  As you can see from the Discussions threads here on github, I'm pretty much the only one asking questions so it feels like I'm monopolizing the discussions and bugging those who do support it. I really dislike having to post so many questions and bothering you and others. But there are explanations that are missing and not sure how to get answers. Like you said, if I knew what the answers were, I'd likely post requests for documentation updates. But about all I could request is "can someone explain x" and perhaps that is sufficient to trigger an update. As for my still lengthening list of questions, do you really want me to post them all or put them in one post?

**Reply by dbelanger** [reply]
You're not alone Bruce, we've all felt like we're monopolizing the discussions at some point in time.  This usually starts at the beginning of taking up a framework or new technology.  It's never a bother and the whole point of this discussion group is so that questions can be answered for future users and so tickets can be created directly to the ticketing system in order to grow the framework.

Version 3 is in the works but naturally the process is much slower as this is a small community.  Since version 2, Peter has been gracious enough to offer corporate sponsorship as well as starting an [Open Collective account](https://opencollective.com/cfwheels) so I'd say the project is still very much alive and the more people that use it, the faster it will evolve!

**Comment by timsayshey** [comment]
Just popping in. Huge long time fan of CFWheels since I've built open source projects on it, I built my career on it, and also now my SaaS company. When I started with it nearly 10ish years ago, the community was also struggling as the original team had moved on, development had completely stopped (long story), then there was a renaissance period where Tom King and others relaunched it, which brought about a stable version 2, new brand, new life, and brings us to where we are today. I'd say it's probably mostly in maintenance mode but these things kind of ebb and flow. Huge props to the team still chugging along working on it though. In regards to the aging docs, I understand that and have had the same issue from time to time, and have had questions go unanswered here on Github. What did help though was using ChatGPT, I'm actually super surprised at how well it understands CFWheels and can write code for it and answer questions about conventions, etc. As far as community goes, the CFML Slack had a CFWheels channel that used to be super helpful, hopefully someone in the community could restart that (maybe with some ground rules so that it stays a friendly place as it should be). Anyway, hope that helps!
